### PR TITLE
sensing: sensing.h: fix doxygen warnings

### PR DESCRIPTION
**Pull request by lixuzha**
Add missing parameter information in `sensing_data_event_t`.

### CODE REVIEW

File: include/zephyr/sensing/sensing.h [hunk 1]
```diff
@@ -150,7 +152,7 @@ struct sensing_sensor_info {
  */
 struct sensing_callback_list {
 	sensing_data_event_t on_data_event;
-	void *context;
+	void *context; /**< Associated context with on_data_event */
```
Comment (kartben) [inline include/zephyr/sensing/sensing.h:155]: Thanks @lixuzha! When pointing out the missing doxygen comments in the recently added fields, I was also hoping you could do a full pass on the various sensing headers :) For example, it probably doesn't make sense to only document `context` in this struct, and not `on_data_event` at the line just above.

If this helps, you may check https://docs.zephyrproject.org/api-coverage/latest/zephyr/sensing/index.html

Comment (lixuzha) [reply include/zephyr/sensing/sensing.h]: Fixed. Thanks for @qianruh 's help.

**Comment by MaureenHelm** [comment]
@kartben can you revisit this please?

**Comment by lixuzha** [comment]
Rebased the PR to latest code.

**Comment by lixuzha** [comment]
@kartben Could you help to review this please?

**Review by teburd — DISMISSED** [review]

File: include/zephyr/sensing/sensing_sensor.h [hunk 2]
```diff
@@ -93,57 +100,101 @@ struct sensing_connection {
...
 		.vendor = DT_NODE_VENDOR_OR(node, NULL),		\
 		.model = DT_NODE_MODEL_OR(node, NULL),			\
-		.minimal_interval = DT_PROP(node, minimal_interval),\
+		.minimal_interval = DT_PROP(node, minimal_interval),	\
 	};
 
+/**
+ * @brief Macro to generate a name for a connections array.
+ *
+ * This macro generates a name for a connections array based on a node.
```
Comment (kartben) [inline include/zephyr/sensing/sensing_sensor.h:150]: would be nice to actually add the `@param`s entries for all the macros

File: include/zephyr/sensing/sensing_sensor.h [hunk 3]
```diff
@@ -153,20 +204,47 @@ extern struct sensing_sensor SENSING_SENSOR_SOURCE_NAME(idx, node);	\
...
+	enum sensor_channel chan; /**< Channel of the sensor submit configuration. */
+	const int info_index;     /**< Info index of the sensor submit configuration. */
+	const bool is_streaming;  /**< Streaming flag of the sensor submit configuration. */
 };
 
+/**
+ * @brief External declaration for the sensing I/O device API.
+ *
+ * This external declaration represents the sensing I/O device API.
+ */
```
Comment (kartben) [inline include/zephyr/sensing/sensing_sensor.h:223]: see earlier comment re: internal. Do not hesitate to put internal stuff inside `@cond INTERNAL_HIDDEN` blocks. You may be more concise with the documentation for these (and don't even need to use doxygen syntax) as they're not meant for external consumption

**Review by kartben — CHANGES_REQUESTED** [review]
thanks a lot for adding documentation here! 
General comment is that some of the comments are not really providing explanation/context but O would be happy if only the "suggested changes" were addressed for now ; the rest can be improved at a later date :)

Thanks again!

**Comment by lixuzha** [comment]
> thanks a lot for adding documentation here! General comment is that some of the comments are not really providing explanation/context but O would be happy if only the "suggested changes" were addressed for now ; the rest can be improved at a later date :)
> 
> Thanks again!

Ok, thanks. I'll address them.

File: include/zephyr/sensing/sensing_sensor.h [hunk 4]
```diff
@@ -93,57 +100,101 @@ struct sensing_connection {
...
-	struct k_timer timer;
-	struct rtio_sqe *stream_sqe;
-	atomic_t flag;
-	struct sensing_connection *conns;
+	enum sensing_sensor_state state;  /**< State of the sensor instance. */
+	struct rtio_iodev *iodev;         /**< An RTIO device info of the sensor instance. */
+	struct k_timer timer;             /**< Timer of the sensor instance. */
+	struct rtio_sqe *stream_sqe;      /**< Stream sqe of the sensor instance. */
+	atomic_t flag;                    /**< Flag of the sensor instance. */
+	struct sensing_connection *conns; /**< Connections of the sensor instance. */
```
[the remark this answers is not in the document]
Comment (lixuzha) [reply include/zephyr/sensing/sensing_sensor.h:118]: It's for internal use. I'll put them inside @cond INTERNAL_HIDDEN blocks. Thanks.

File: include/zephyr/sensing/sensing_sensor.h (hunk 3 above)
Comment (lixuzha) [reply include/zephyr/sensing/sensing_sensor.h:223]: Ok, thanks.

File: include/zephyr/sensing/sensing.h [hunk 5]
```diff
@@ -151,41 +157,40 @@ struct sensing_sensor_info {
...
+ *
+ * This API just returns read only information of sensor instances, pointer info will
+ * directly point to internal buffer, no need for caller to allocate buffer,
+ * no side effect to sensor instances.
+ *
+ * @param num_sensors Get number of sensor instances.
+ *
+ * @param info For receiving sensor instances' information array pointer.
+ *
+ * @return 0 on success or negative error value on failure.
```
[the remark this answers is not in the document]
Comment (lixuzha) [reply include/zephyr/sensing/sensing.h]: Fixed.

File: include/zephyr/sensing/sensing_datatypes.h [hunk 6]
```diff
@@ -86,11 +87,12 @@ struct sensing_sensor_value_3d_q31 {
  * uint32_t version
  */
 struct sensing_sensor_value_uint32 {
+	/**< Header of the sensor value data structure. */
```
[the remark this answers is not in the document]
Comment (lixuzha) [reply include/zephyr/sensing/sensing_datatypes.h]: Fixed.

File: include/zephyr/sensing/sensing_sensor.h [hunk 7]
```diff
@@ -57,30 +57,37 @@ struct sensing_sensor_register_info {
...
 struct sensing_connection {
-	sys_snode_t snode;
-	struct sensing_sensor *source;
-	struct sensing_sensor *sink;
-	/* interval and sensitivity set from client(sink) to reporter(source) */
-	uint32_t interval;
+	sys_snode_t snode;             /**< Node in the singly-linked list of connections. */
+	struct sensing_sensor *source; /**< Source sensor of the connection. */
+	struct sensing_sensor *sink;   /**< Sink sensor of the connection. */
+	uint32_t interval;             /**< Interval of the connection. */
```
[the remark this answers is not in the document]
Comment (lixuzha) [reply include/zephyr/sensing/sensing_sensor.h]: Added.

File: include/zephyr/sensing/sensing_sensor.h (hunk 2 above)
Comment (lixuzha) [reply include/zephyr/sensing/sensing_sensor.h:158]: Added.

File: include/zephyr/sensing/sensing.h [hunk 8]
```diff
@@ -38,16 +38,20 @@ extern "C" {
  */
 struct sensing_sensor_version {
 	union {
-		uint32_t value;
+		uint32_t value; /** The version represented as a 32-bit value. */
```
Comment (kartben) [inline include/zephyr/sensing/sensing.h:41]: Not sure why all the `/**< ...  **/` style comments are gone: you need them when you want to indicate the comment applies to the compilation unit that's *before* (hence `<`) the comment

**Review by kartben — CHANGES_REQUESTED** [review]

File: include/zephyr/sensing/sensing.h (hunk 8 above)
Comment (lixuzha) [reply include/zephyr/sensing/sensing.h]: Updated.

**Review by kartben — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**